src/overlays/actors/ovl_En_Si/z_en_si.c

```diff
@@ -77,24 +90,81 @@ static DamageTable D_8098CDB8 = {
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Si/func_8098CB70.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Si/func_8098CBDC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Si/EnSi_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Si/EnSi_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Si/EnSi_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Si/EnSi_Draw.s")
+void func_8098CA20(EnSi* this, PlayState* play) {
+    this->collider_sphere.dim.worldSphere.center.x = (s16)(s32)this->actor.world.pos.x;
+    this->collider_sphere.dim.worldSphere.center.y = (s16)(s32)this->actor.world.pos.y;
+    this->collider_sphere.dim.worldSphere.center.z = (s16)(s32)this->actor.world.pos.z;
+    this->collider_sphere.dim.worldSphere.radius =
+        (s16)(s32)((f32)this->collider_sphere.dim.modelSphere.radius * this->collider_sphere.dim.scale);
+    if ((s32)this->actor.colChkInfo.health > 0) {
+        CollisionCheck_SetAC(play, &play->colChkCtx, &this->collider_sphere.base);
+    }
+    CollisionCheck_SetOC(play, &play->colChkCtx, &this->collider_sphere.base);
+}
+
+void func_8098CAD0(EnSi* this, PlayState* play) {
+    s32 temp_a2;
+
+    temp_a2 = (s32)(this->actor.params & 0xFC) >> 2;
+    if ((temp_a2 < 0x20) && (temp_a2 >= 0)) {
+        Flags_SetTreasure(play, temp_a2);
+    }
+    Item_Give(play, ITEM_SKULL_TOKEN);
+    if (Inventory_GetSkullTokenCount(play->sceneId) >= 0x1E) {
+        Message_StartTextbox(play, 0xFCU, NULL);
+        Audio_PlayFanfare(NA_SE_PL_DUMMY_290);
+        return;
+    }
+    Message_StartTextbox(play, 0x52U, NULL);
+    Audio_PlayFanfare(NA_BGM_GET_SMALL_ITEM);
+}
+
+void func_8098CB70(EnSi* this, PlayState* play) {
+    if ((this->actor.flags & 0x2000) == 0x2000) {
+        this->actionFunc = func_8098CBDC;
+        goto block_4;
+    }
+    if (this->collider_sphere.base.ocFlags2 & 1) {
+        func_8098CAD0(this, play);
+        Actor_Kill(&this->actor);
+        return;
+    }
+block_4:
+    this->actor.shape.rot.y += 0x38E;
+}
+
+void func_8098CBDC(EnSi* this, PlayState* play) {
+    if ((this->actor.flags & 0x2000) != 0x2000) {
+        func_8098CAD0(this, play);
+        Actor_Kill(&this->actor);
+    }
+}
+
+void EnSi_Init(Actor* thisx, PlayState* play) {
+    EnSi* this = THIS;
+
+    Collider_InitSphere(play, &this->collider_sphere);
+    Collider_SetSphere(play, &this->collider_sphere, &this->actor, &sSphereInit);
+    CollisionCheck_SetInfo2(&this->actor.colChkInfo, &sDamageTable, &sColChkInfoInit);
+    Actor_SetScale(&this->actor, 0.25f);
+    this->actionFunc = func_8098CB70;
+}
+
+void EnSi_Destroy(Actor* thisx, PlayState* play) {
+    EnSi* this = THIS;
+    Collider_DestroySphere(play, &this->collider_sphere);
+}
+
+void EnSi_Update(Actor* thisx, PlayState* play) {
+    EnSi* this = THIS;
+    this->actionFunc(this, play);
+    func_8098CA20(this, play);
+    Actor_SetFocus(&this->actor, 0.0f);
```

Comment:
```suggestion
    EnSi* this = THIS;
    
    this->actionFunc(this, play);
    func_8098CA20(this, play);
    Actor_SetFocus(&this->actor, 0.0f);
```
Format wise, we like to keep a space between the variables and the general body of the function.